src/rpc/rawtransaction.cpp

```diff
@@ -1270,6 +1286,123 @@ static std::string WriteHDKeypath(std::vector<uint32_t>& keypath)
 }
 
+UniValue blindpsbt(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 2)
+        throw std::runtime_error(
+            "blindpsbt \"psbt\" ( ignoreblindfail )\n"
+            "\nUse the blinding data from the PSBT inputs to generate the blinding data for the PSBT outputs.\n"
+
+            "\nArguments:\n"
+            "1. \"psbt\"            (string, required) The PSBT base64 string\n"
+            "2. ignoreblindfail   (bool, optional, default=true) Return a transaction even when a blinding attempt fails due to number of blinded inputs/outputs.\n"
+
+            "\nResult:\n"
+            "  \"psbt\"          (string) The base64-encoded partially signed transaction\n"
+            "\nExamples:\n"
+            + HelpExampleCli("blindpsbt", "\"psbt\" ignoreblindfail")
+        );
+
+    RPCTypeCheck(request.params, {UniValue::VSTR, UniValue::VBOOL}, true);
+
+    // Unserialize the transactions
+    PartiallySignedTransaction psbtx;
+    std::string error;
+    if (!DecodePSBT(psbtx, request.params[0].get_str(), error)) {
+        throw JSONRPCError(RPC_DESERIALIZATION_ERROR, strprintf("TX decode failed %s", error));
+    }
+
+    bool fIgnoreBlindFail = true;
+    if (!request.params[1].isNull()) {
+        fIgnoreBlindFail = request.params[1].get_bool();
+    }
+
+    // TODO(gwillen): Refactor out significant duplicated code between here and rawblindrawtransaction.
+
+    std::vector<CAmount> input_amounts;
+    std::vector<uint256> input_blinds;
+    std::vector<uint256> input_asset_blinds;
+    std::vector<CAsset> input_assets;
+    std::vector<uint256> output_value_blinds;
+    std::vector<uint256> output_asset_blinds;
+    std::vector<CAsset> output_assets;
+    std::vector<CPubKey> output_pubkeys;
+
+    int n_blinded_ins = 0;
+
+    // TODO(gwillen): If blinding is not possible due to missing input data, we should bail here with a useful error message.
+    for (const auto& input : psbtx.inputs) {
+        input_blinds.push_back(input.value_blinding_factor);
+        input_asset_blinds.push_back(input.asset_blinding_factor);
+        input_assets.push_back(input.asset);
+        input_amounts.push_back(input.value ? *input.value : CAmount(-1));
+
+        if (!input_blinds.back().IsNull()) {
+            n_blinded_ins++;
+        }
+    }
+
+    for (auto& output : psbtx.outputs) {
+        output_pubkeys.push_back(output.blinding_pubkey);
+    }
+
+    // How many are we trying to blind?
+    int num_pubkeys = 0;
+    unsigned int keyIndex = -1;
+    for (unsigned int i = 0; i < output_pubkeys.size(); i++) {
+        const CPubKey& key = output_pubkeys[i];
+        if (key.IsValid()) {
+            num_pubkeys++;
+            keyIndex = i;
+        }
+    }
+
+    CMutableTransaction& tx = *psbtx.tx;
+
+    // TODO(gwillen): Replace all this with the 'bonus output' scheme to use an OP_RETURN to balance blinders, with a rangeproof exponent of -1 (public).
```

Comment:
with known blinding pubkey, right?

Reply:
I believe the scheme we talked about -- which I don't expect to fully understand before implementing it -- was to just make up a random pubkey and then throw it away after, because nobody ever needs to be able to rewind these outputs.

Reply:
that works too. Could just be the generator point G or something stupid as well.